x/incentives/keeper/distribute_test.go

```diff
@@ -868,3 +870,196 @@ func (s *KeeperTestSuite) TestGetPoolFromGaugeId() {
 		})
 	}
 }
+
+// TestFunctionalInternalExternalCLGauge is a functional test that covers more complex scenarios relating to distributing incentives through gauges
+// at the end of each epoch.
+//
+//
+// Testing strategy:
+// 1. Initialize variables.
+// 2. Setup CL pool and gauge (gauge automatically gets created at the end of CL pool creation).
+// 3. Create external no-lock gauges for CL pools
+// 4. Create Distribution records to incentivize internal CL no-lock gauges
+// 5. let epoch 1 pass
+//  	- Check that incentive record has been correctly created and gauge has been correctly updated.
+// 		- all perpetual gauges must finish distributing records
+// 6. Remove distribution records for internal incentives using HandleReplacePoolIncentivesProposal
+// 7. let epoch 2 pass
+// 		- check only external non-perpetual gauges with 2 epochs distributed
+// 		- check gauge has been correctly updated
+// 8. let epoch 3 pass
+// 		- nothing distributes as non-perpetual gauges with 2 epochs have ended and perpetual gauges have not been reloaded
+//		- nothing should change interms of incentive records
+func (s *KeeperTestSuite) TestFunctionalInternalExternalCLGauge() {
+	// 1. Initialize variables
+	s.SetupTest()
+
+	requiredBalances := sdk.NewCoins(sdk.NewCoin("eth", sdk.NewInt(1_000_000_000)), sdk.NewCoin("usdc", sdk.NewInt(1_000_000_000)), sdk.NewCoin("uosmo", sdk.NewInt(1_000_000_000)))
+	internalGaugeCoins := sdk.NewCoins(sdk.NewCoin("stake", sdk.NewInt(750_000)))                                               // distributed full sum at epoch
```

Comment:
What is the 1 billion uosmo used for?

Reply:
Bumping this, as I think we need to know exactly why these funds are required before merge

Reply:
sorry for the delay 
that was just an amount so that testaccs[1] and [2] donot run out of funds  because they are creating external incentives. I realized that i was being a little too generous so, i changed it to `defaultExternalGaugeValue` which is just 1M

Reply:
@stackman27 we should only be funding accounts with the amount they intend to use. If we just fund them with some amount, we wont be notified if an account is utilizing more funds than we intended